### DISCUSSION TITLE
Wrapping enum for external rust crate

### DISCUSSION THREAD

**Discussion by JonathanPlasse** [PyO3/pyo3#2787] [Questions]
Is it possible to wrap an enum from an external crate to apply `pyclass` on it?
Thank you.

**Comment by davidhewitt** [comment]
Sorry for the slow reply.

Rust proc-macros work off the syntax they are applied to, so unfortunately you can't just add `#[pyclass]` to "wrap" an external enum.

There are crates like [deflect](https://docs.rs/deflect/latest/deflect/) which enable reflection at runtime, which might eventually provide a way to make this work. However, those techniques are still experimental and only work on Linux, so I'm not rushing to try them out for PyO3 yet.

**Comment by JonathanPlasse** [comment]
Thanks for your response, so I just have to copy the enum and implement a conversion from the copied enum to the external enum.

**Reply by mejrs** [reply]
You need to make a newtype around the enum.

**Reply by JonathanPlasse** [reply]
I tried to make a newtype with
```rust
[pyclass]
struct EnumWrapper(Enum);
```
but the field of the enum is not present I only got an empty class.
